src/screen.rs

```diff
@@ -457,20 +463,20 @@ impl Screen {
             self.desired.add_line();
         }
 
-        let full_line_count = self.desired.cursor.y
+        let full_line_count = self.desired.cursor.y + 1
             - if self.desired.cursor.x == 0
-                && self
-                    .desired
-                    .cursor
-                    .y
-                    .checked_sub(1)
-                    .is_some_and(|y| self.desired.line_datas[y].is_soft_wrapped)
+                && self.desired.cursor.y.checked_sub(1).is_some_and(|y| {
+                    self.desired.line_datas[y].is_soft_wrapped
+                        && self.desired.line_datas[y]
+                            .color_at(self.desired.line_datas[y].len() - 1)
+                            .foreground
+                            == HighlightRole::autosuggestion
```

Comment:
this looks reasonable, might be a small layering violation though.
I wonder if we can improve this future, maybe `is_soft_wrapped` should be an enum with different values for "actual softwrap" or "softwrapped autosuggestion".
This stuff is super complicated..